lib/Experian/IDAuth.pm

```diff
@@ -468,16 +468,19 @@ sub _do_192_authentication {
     $self->_build_request
         || die("Cannot build xml_request for [" . $self->{client_id} . "/$search_option]");
 
-    # Remove old pdf in case this client has done the 192 pdf request before
-    my $file_pdf = $self->_pdf_report_filename;
-    unlink $file_pdf if -e $file_pdf;
-
     eval { $self->_send_request } || do {
         my $err = $@ || '?';
         warn "could not send pdf request: $err";
         return;
     };
 
+    my $result = $self->_xml_as_hash;
+    die "ErrorCode: $result->{ErrorCode}, ErrorMessage: $result->{ErrorMessage}" if $result->{ErrorCode};
+
+    # Remove old pdf in case this client has done the 192 pdf request before
+    my $file_pdf = $self->_pdf_report_filename;
+    unlink $file_pdf if -e $file_pdf;
```

Comment:
I think that should be done in class destructor? I.e. as client finished working with experian the instance of `Experian::IDAuth` should go out of scope and all related files removed.